File: OrmDemo.Dapper/PersonLoader.cs
using System.Collections.Generic;
using System.Linq;
using OrmDemo.Domain;
using System.Data.SqlClient;
using Dapper;

namespace OrmDemo.Dapper
{
    public class PersonLoader
    {
        public static IEnumerable<Person> LoadPersons(IEnumerable<string> names)
        {
            const string connectionString =
                @"Data Source=(localdb)\MSSQLLocalDB;Initial Catalog=OrmDemo;Integrated Security=True;Connect Timeout=30;Encrypt=False;TrustServerCertificate=False;ApplicationIntent=ReadWrite;MultiSubnetFailover=False";
            const string sql = @"select * from people p
                            join organizations o on p.MemberOf_Id = o.Id
                            where p.Name = @name";

            using (var connection = new SqlConnection(connectionString))
            {
                connection.Open();
                foreach (var name in names)
                {
                    yield return connection.Query<Person, Organization, Person>(sql,
                        (person, org) => { person.MemberOf = org; return person; },
                        new { name })
                        .First();
                }
                connection.Close();
            }
        }
    }
}

File: OrmDemo.Fsharp/PersonLoader.fs
module OrmDemo.Fs.PersonLoader

open FSharp.Data
open OrmDemo.Domain

[<Literal>]
let ConnectionString = 
    @"Data Source=(localdb)\MSSQLLocalDB;Initial Catalog=OrmDemo;Integrated Security=True;Connect Timeout=30;Encrypt=False;TrustServerCertificate=False;ApplicationIntent=ReadWrite;MultiSubnetFailover=False"

let LoadPersons (names:seq<string>) =
   let createPerson id name oId oName amount =
      let toString =
         function 
         | Some s -> s
         | None -> ""

      Person(id, (toString name), Organization(oId, (toString oName), amount))

   use cmd = new SqlCommandProvider<"PersonQuery.sql", ConnectionString, ResolutionFolder = "..\OrmDemo\sql">()
   
   seq{
      for r in cmd.Execute() do
         yield createPerson r.Id r.Name r.orgId r.oName r.Amount
   }
